language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -459,26 +459,31 @@ object DottyLanguageServer {
         CIK.Field
     }
 
-    val label = sym.name.show.toString
+    val label = sym.name.show
     val item = new lsp4j.CompletionItem(label)
-    item.setDetail(sym.info.widenTermRefExpr.show.toString)
+    item.setDetail(sym.info.widenTermRefExpr.show)
     item.setKind(completionItemKind(sym))
     item
   }
 
-  private def docMarkedStrings(comment: Option[Comment], typeInfo: String): List[lsp4j.MarkedString] = {
-
-    val docHover = comment.map { comment =>
-      new lsp4j.MarkedString("scala", comment.raw)
-    }
-
-    val typeInfoHover = new lsp4j.MarkedString()
-    typeInfoHover.setValue(typeInfo)
-
-    typeInfoHover :: docHover.toList
+  private def hoverContent(typeInfo: String, comment: Option[Comment]): lsp4j.MarkupContent = {
+    val markup = new lsp4j.MarkupContent
+    markup.setKind("markdown")
+    markup.setValue((
+      comment.map(_.raw) match {
+        case Some(comment) =>
+          s"""```scala
+             |$typeInfo
+             |$comment
+             |```"""
```

Comment:
There used to be a thin line that separated the type info from the comment. It would be nice to re-introduce it, since it makes the hover information easier to read:

```scala
case Some(comment) =>
  s"""```scala
     |$typeInfo
     |```
     |---
     |```scala
     |$comment
     |```"""
```

Reply:
I tried that but that's not equivalent to the thin line we had before: it adds a really thick line which I find distracting

Reply:
There's an issue for this: https://github.com/Microsoft/language-server-protocol/issues/518